nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -5,17 +5,22 @@ with lib;
 let
   cfg = config.services.matrix-synapse;
   logConfigFile = pkgs.writeText "log_config.yaml" cfg.logConfig;
+  mkResource = r: ''{names: [${concatStringsSep "," r.names}], compress: ${if r.compress then "true" else "false"}}'';
+  mkListener = l: ''{port: ${l.port}, bind_address: ${l.bind_address}, type: ${l.type}, tls: ${if l.tls then "true" else "false"}, x_forwarded: ${if l.x_forwarded then "true" else "false"}, resources: [${concatStringsSep "," (map mkResource l.resources)}]}'';
```

Comment:
port has to be a string here. That kinda sucks :( . It'd be nice if we could either do an if/else on the type and convert if necessary, or enforce port to be a number somehow.